initializr-generator/src/main/java/io/spring/initializr/generator/language/java/JavaTypeDeclaration.java

```diff
@@ -26,11 +26,14 @@
  *
  * @author Andy Wilkinson
  * @author Matt Berteaux
+ * @author Yifan Li
  */
 public class JavaTypeDeclaration extends TypeDeclaration {
 
 	private int modifiers;
 
+	private final List<String> implementsInterfaceNames = new ArrayList<>();
```

Comment:
An interface may be a complex type with generic that should themselves be imported.

Reply:
Well, this is a problem. But currently the extend is also cannot support this kind of situation. Judging by the current implementation, i think this requires relatively large changes. Is it possible to set up another pr to solve the problem together?

By the way. I haven't thought of a very good solution at the moment. Is it possible to provide an API to let the caller specify the import manually?

Reply:
> Is it possible to set up another pr to solve the problem together?

Sorry, I didn't get that.

There is no API to import things explicitly. I don't think we should provide one, as it is too much of a compromise for something you should be able to express via the API.

Reply:
ok, May be due to language issues, I did not express my thoughts clearly.

I think the current process has the following problems:
1. Classes with the same name are not supported (different class paths): the current processing logic of `getUnqualifiedName` will cause reference confusion in the generated code.

2. Does not support generics: The current `determineImports` method cannot handle naming with generics, resulting in generics when importing

3. I think the current processing logic is too complex and leads to inflexibility. Can not meet the code generation in many scenarios, such as: initialize the object in the method, call the constructor ...

My current solution is as follows:
1. Provide `List <String> imports` in` TypeDeclaration` to support manually specifying the classes that need to be import in the generated code, to avoid the current complex and imperfect program processing. Specific examples are as follows:
```java
public class TypeDeclaration implements Annotatable {
...

private final List <String> imports = new ArrayList <> ();

public void withImport (String name) {
    this.imports.add (name);
}

public List <String> getImports () {
    return this.imports;
}

}
```

2. Define the `JavaCommonExpression` class and provide manual entry of code lines to solve the problem of flexibility in code generation
```java
public class JavaCommonExpression extends JavaExpression {

private List <String> statementLines = new ArrayList <> ();

public void addStatement (String line) {
    this.statementLines.add (line);
}

public List <String> getStatements () {
    return this.statementLines;
}

}

// JavaSourceCodeWriter for this type of output
private void writeCommonStatement (IndentingWriter writer, JavaCommonExpression expression) {
    expression.getStatements (). stream (). filter ((str)-> str.length ()> 0) .forEach (writer :: println);
}
```

3. Modify `determineImports` to intercept the class name before the` <`symbol and import name
```java
...
imports.replaceAll ((importLine)-> {
      if (importLine.contains ("<")) {
            return importLine.substring (0, importLine.indexOf ("<"));
      }
     else {
             return importLine;
     }
});
Collections.sort (imports);
return new LinkedHashSet <> (imports);
```

@snicoll  Is the above solution feasible？:alien: